### DISCUSSION TITLE
Querying property graph using SPARQL

### DISCUSSION THREAD

**Discussion by qwaider** [aws/graph-explorer#476] [Q&A] [answered]
Dears,

I deployed GraphExplorer that worked perfectly .. and at some point wanted to try the sparql endpoint .. there were I saw it doesn't work and it gives 0 counters for subject and predicate.. while the neptune graph has +6M Nodes and Edges.

So the question is if I can query the property graph using sparql or not?
Most of the documentations and chatgpt gives the intention answer "yes" except this one: https://docs.aws.amazon.com/neptune/latest/userguide/get-started-access-graph.html 
which says it clearly "no".

What's your experience .. since also from this documentation: https://docs.aws.amazon.com/neptune/latest/userguide/feature-overview-data-model.html
It sounds like enabling the indexing of the triples would allow RDFs to be created at populating time then can be queried using sparql.

Thanks for your insights.
Regards,

**Comment by triggan** [comment, marked as answer]
A Neptune Database cluster can store Labeled Property Graph and RDF data in the same cluster, but they are stored separately.  Labeled Property Graph data can be written/read using either Gremlin or openCypher.  RDF data can be written/read using SPARQL.  But you cannot read RDF data using the LPG languages, nor can you read LPG data using SPARQL.   As such, you cannot use Graph Explorer doing the same.

**Reply by qwaider** [reply]
Thanks a lot for the info provided .. Since your knowledge on the question: Is there an easy/automatised way to sync property graph data to RDF's as it's inserted/updated/removed?

Thanks again

**Reply by qwaider** [reply]
@triggan
